provisioning_templates/snippet/create_users.erb

```diff
@@ -0,0 +1,25 @@
+<%#
+name: create_users
+kind: snippet
+snippet: true
+-%>
+<%- users = @host.owner_type == 'Usergroup' ? @host.owner.all_users : [@host.owner] -%>
+<%- users.each do |user| -%>
+<%=   "useradd #{user.login}" %>
+<%-   if user.ssh_authorized_keys && user.ssh_authorized_keys.any? -%>
+<%=     "mkdir -p -m 0700 #{user.login}/.ssh" %>
+<%=     "cat >> ~#{user.login}/.ssh/authorized_keys <<EOF" %>
+<%-     index = 0 -%>
+<%-     user.ssh_keys.each do |key| -%>
+<%-       if index == 0 -%>
+<%=        "#{key.type} #{key.key} #{key.comment}" %>
+<%-       else -%>
+<%=        "#{key.type} #{key.key} #{key.comment} - #{index}" %>
```

Comment:
I'm not getting that line... should be index be subtracted or printed? if the later, why?